### PR TITLE
Release v0.17.0

### PR DESCRIPTION
**Pull request by Jarema**

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,42 @@
...
+* Refactor of JetStream API
+* Improvements of JetStream internals
+* Key-Value Store support
+* Object Store support
+
+## Breaking Changes
+* Introduce a `JetStream` type by @caspervonb in https://github.com/nats-io/nats.rs/pull/247
+* Move Consumer Management to JetStream by @Jarema in https://github.com/nats-io/nats.rs/pull/250
+* Re-work JetStream consumer interface by @caspervonb in https://github.com/nats-io/nats.rs/pull/252
+* Rename create_stream to add_stream by @Jarema in https://github.com/nats-io/nats.rs/pull/251
```
Comment (caspervonb) [inline CHANGELOG.md:14]:
```suggestion
* Rename create_stream to add_stream by @Jarema in https://github.com/nats-io/nats.rs/pull/251
* Change return type of add_consumer to `ConsumerInfo` by @caspervonb in https://github.com/nats-io/nats.rs/pull/252
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,3 +1,42 @@
...
+
+## Added
+* Add header module by @caspervonb in https://github.com/nats-io/nats.rs/pull/260
+* Implement key-value store by @caspervonb in https://github.com/nats-io/nats.rs/pull/267
+* Implement object store by @caspervonb in https://github.com/nats-io/nats.rs/pull/269
+* Lame duck mode support by @Jarema in https://github.com/nats-io/nats.rs/pull/265
+* Add domain field to `PubAck` by @caspervonb in https://github.com/nats-io/nats.rs/pull/243
+* Add support for JetStream publishing by @caspervonb in https://github.com/nats-io/nats.rs/pull/248
+* Add `error_callback` by @derekcollison in https://github.com/nats-io/nats.rs/pull/253
+
```
Comment (caspervonb) [inline CHANGELOG.md:24]:
```suggestion
* Add `get_message` to JetStream context by @caspervonb in https://github.com/nats-io/nats.rs/pull/267
* Add `get_last_message` to JetStream context  by @caspervonb in https://github.com/nats-io/nats.rs/pull/267
```

Comment (caspervonb) [reply CHANGELOG.md]:
```suggestion
* New interfaces and methods for push based subscriptions by @caspervonb in https://github.com/nats-io/nats.rs/pull/252
```

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,3 +1,44 @@
...
+* Add header module by @caspervonb in https://github.com/nats-io/nats.rs/pull/260
+* Implement key-value store by @caspervonb in https://github.com/nats-io/nats.rs/pull/267
+* Implement object store by @caspervonb in https://github.com/nats-io/nats.rs/pull/269
+* Lame duck mode support by @Jarema in https://github.com/nats-io/nats.rs/pull/265
+* Add domain field to `PubAck` by @caspervonb in https://github.com/nats-io/nats.rs/pull/243
+* Add support for JetStream publishing by @caspervonb in https://github.com/nats-io/nats.rs/pull/248
+* Add `error_callback` by @derekcollison in https://github.com/nats-io/nats.rs/pull/253
+* Add `get_message` to JetStream context by @caspervonb in https://github.com/nats-io/nats.rs/pull/267
+* Add `get_last_message` to JetStream context  by @caspervonb in https://github.com/nats-io/nats.rs/pull/267
+## Changed
```
Comment (caspervonb) [inline CHANGELOG.md:27]: My bad, formatting and GitHub UI :D

```suggestion

## Changed
```

File: CHANGELOG.md [hunk 4]
```diff
@@ -1,3 +1,45 @@
+# 0.17.0
+## Overview
+
+This release brings a lot of changes and refactors among which the highlights are:
+* Refactor of JetStream API
```
Comment (caspervonb) [inline CHANGELOG.md:5]:
```suggestion
* A complete rewrite of the JetStream API with a new subscription interface
```

Comment (caspervonb) [reply CHANGELOG.md]: Rewrite rather than refactor? As refactor kind of implies same level of functionality, we support a lot more now :)

**Review by caspervonb — APPROVED** [review]

File: README.md [hunk 5]
```diff
@@ -107,6 +107,37 @@ nc.publish_request("foo", &reply, "Help me!")?;
...
+let js = nats::jetstream::new(nc);
+
+// add_stream converts a str into a
+// default `StreamConfig`.
+js.add_stream("my_stream")?;
+```
+
+Create a new consumer:
+```rust
+let nc = nats::connect("my_server::4222")?;
```
Comment (derekcollison) [inline README.md:123]: Put `demo.nats.io` here, it supports JetStream now and is a good placeholder.

File: README.md [hunk 6]
```diff
@@ -107,6 +107,37 @@ nc.publish_request("foo", &reply, "Help me!")?;
...
+js.add_stream("my_stream")?;
+js.add_consumer("my_stream", "my_consumer")?;
+```
+
+Create a new subscription:
+```rust
+let nc = nats::connect("my_server::4222")?;
+let js = nats::jetstream::new(nc);
+
+js.add_stream("my_stream")?;
```
Comment (derekcollison) [inline README.md:135]: Show example with options?

**Review by derekcollison — COMMENTED** [review]
Just a few comments, looks really good!

**Pull request merged**